File: WEB/nexa/app/Models/EpiModel.php
<?php

namespace App\Models;

use CodeIgniter\Model;

class EpiModel extends Model
{
    protected $table = 'EPI';

    protected $primaryKey = 'ID';

    protected $allowedFields = [
        'NOME_EPI',
        'IMAGEM_EPI',
        'DESCRICAO_EPI',
        //'FK_CPF_FUNCIONARIO'
    ];

    protected $returnType = 'array';

   protected $validationRules = [

    'NOME_EPI' => [
        'rules' => 'required',
        'errors' => [
            'required' => 'O nome do EPI é obrigatório.'
        ]
    ],

    'DESCRICAO_EPI' => [
        'rules' => 'required',
        'errors' => [
            'required' => 'A descrição é obrigatória.'
        ]
    ]

];
}

File: Mobile/nexa_app/lib/models/epi_model.dart
class EpiModel {
  final int id;
  final String nome;
  final String imagem;
  final String descricao;

  EpiModel({
    required this.id,
    required this.nome,
    required this.imagem,
    required this.descricao,
  });

  factory EpiModel.fromJson(Map<String, dynamic> json) {
    return EpiModel(
      id: int.tryParse(
            json['ID']?.toString() ??
                json['id']?.toString() ??
                '0',
          ) ??
          0,
      nome: json['NOME_EPI']?.toString() ??
          json['nome']?.toString() ??
          '',
      imagem: json['IMAGEM_EPI']?.toString() ??
          json['imagem']?.toString() ??
          '',
      descricao: json['DESCRICAO_EPI']?.toString() ??
          json['descricao']?.toString() ??
          '',
    );
  }
}